PYME/IO/DataSources/SupertileDatasource.py

```diff
@@ -38,24 +38,36 @@
 
 class DataSource(BaseDataSource):
     moduleName = 'SupertileDataSource'
-    def __init__(self, filename, taskQueue=None):
-        from PYME.Analysis.tile_pyramid import ImagePyramid
-        self.tile_base, query = filename.split('?')
-        qp = parse_qs(query)
-        self.level = int(qp.get('level', [0])[0])
-        self.stride = int(qp.get('stride', [3])[0])
-        self.overlap = int(qp.get('overlap', [1])[0])
+    def __init__(self, pyramid, level=0, stride=3, overlap=1):
```

Comment:
I'll try and re-work this. 

For future reference, a `DataSourceModule.DataSource` **must**  accept a filename in the constructor (this is, amongst other things, related to how localisation analysis can work for file types other than .h5 and .pcs). The solution in this case is probably to use inheritance rather than a static method - i.e. have `SupertileDatasource.SupertileDataSource` with your new constructor, and `SupertileDatasource.DataSource` as an over-ridden version (or potentially even just a function which calls the static method).

Reply:
Actually, just realised that the `from_tile_series(...)` static method does something completely different to what the initial constructor does - namely it creates a pyramid from scratch, rather than loading an existing pyramid.